tests/junit_common.py

```diff
@@ -312,6 +312,47 @@ def diff_cgns(test_cgns: Path, true_cgns: Path, tolerance: float = 1e-12) -> str
     return proc.stderr.decode('utf-8') + proc.stdout.decode('utf-8')
 
 
+def test_case_output_string(test_case: TestCase, spec: TestSpec, mode: RunMode,
```

Comment:
This is a pretty big change to the common JUNIT code - have you tested that this core file can drop into Ratel with these changes? @zatkins-dev comments?

Reply:
I'll review/test this afternoon with Ratel

Reply:
This looks like a pretty clean refactor of the output string generation, I can't imagine it causing any issues. I'll double check but tentatively looks good

Reply:
LGTM! Ratel tests ran without issue locally. If this is finalized, I can make a matching MR in Ratel for the changes.